dwave/optimization/_model.pyx

```diff
@@ -251,6 +233,37 @@ cdef class _Graph:
 
         return model
 
+    @staticmethod
+    def _from_file_header(file):
+        """Read/validate the header from a model file.
+
+        Advances the stream position to the end of the header.
+
+        Returns:
+            The serialization version as a 2-tuple.
+
+            The header data as a dict.
+        """
+        prefix = b"DWNL"
+
```

Comment:
I'd assert at the beginning the minimal length to be `len(prefix) + 2 + 4`, or fail with `ValueError`.